odoo/addons/test_read_group/tests/test_m2m_grouping.py

```diff
@@ -0,0 +1,228 @@
+            groupby=['user_ids'],
+        )
+        self.assertEqual(as_admin, [
+            {
+                'user_ids': users[0].id,
+                'user_ids_count': 1,
+                'name': ['Find Yes Man'],
+                '__domain': [('user_ids', '=', users[0].id)],
+            },
+            {
+                'user_ids': users[1].id,
+                'user_ids_count': 1,
+                'name': ['Find Yes Man'],
+                '__domain': [('user_ids', '=', users[1].id)],
+            },
+        ])
+
+        # ir.rule should apply
+        task = task.with_user(self.browse_ref('base.user_demo'))
+
+        # warming up various caches; this will avoid queries
+        task.read_group(domain=[], fields=['name:array_agg'], groupby=['user_ids'])
+
+        expected = """
+            SELECT
+                min("test_read_group_task".id) AS id,
+                count("test_read_group_task".id) AS "user_ids_count",
+                array_agg("test_read_group_task"."name") AS "name",
+                "test_read_group_task__test_read_group_task_user_rel__user_ids"."id" as "user_ids"
+            FROM "test_read_group_task"
+            LEFT JOIN "test_read_group_task_user_rel"
+                AS "test_read_group_task__test_read_group_task_user_rel"
+                ON ("test_read_group_task"."id"
+                    = "test_read_group_task__test_read_group_task_user_rel"."task_id")
+            LEFT JOIN "test_read_group_user"
+                AS "test_read_group_task__test_read_group_task_user_rel__user_ids"
+                ON ("test_read_group_task__test_read_group_task_user_rel"."user_id"
+                    = "test_read_group_task__test_read_group_task_user_rel__user_ids"."id"
+                    AND
+                    "test_read_group_task__test_read_group_task_user_rel__user_ids"."id"
+                    IN (
+                        SELECT "test_read_group_user".id
+                        FROM "test_read_group_user"
+                        WHERE
+                            ("test_read_group_user"."name" = %s)
+                    ))
+            GROUP BY "test_read_group_task__test_read_group_task_user_rel__user_ids"."id"
+            ORDER BY "user_ids"
+        """
+        with self.assertQueries([expected]):
+            as_demo = task.read_group(
+                domain=[],
+                fields=['name:array_agg'],
+                groupby=['user_ids'],
+            )
+        self.assertEqual(as_demo, [
+            {
+                'user_ids': users[0].id,
+                'user_ids_count': 1,
+                'name': ['Find Yes Man'],
+                '__domain': [('user_ids', '=', users[0].id)],
+            },
+            {   # the second one appears as assigned to nobody
+                'user_ids': False,
+                'user_ids_count': 1,
+                'name': ['Find Yes Man'],
+                '__domain': [('user_ids', '=', False)],
+            },
```

Comment:
I did not expect this.

Reply:
This is what we discussed privately last time, I thought it would be best not to show it **at all** but you told me that it made sense for it to appear as `NULL` / `False`, or at least that's how I interpreted what you told me.

Or am I misunderstanding what you expect here?